tests/utils/jwtUtil.js

```diff
@@ -0,0 +1,169 @@
+      'info attached', (done) => {
+      const request = {
+        headers: { },
+        session: { },
+      };
+      request.headers.authorization = userToken;
+      jwtUtil.verifyToken(request, dummyCallback)
+      .then(() => {
+        expect(request.headers.userName).to.equal(userInst.name);
+        expect(request.headers.profileName).to.equal('___myProfile');
+        expect(request.headers.tokenName).to.equal('___myRefocusUser');
+        expect(request.headers.isAdmin).to.equal(false);
+        expect(request.headers.isBot).to.equal(false);
+        expect(request.headers.isCollector).to.equal(false);
+        return done();
+      }).catch(done);
+    });
+
+    it('verifyCollectorToken and make sure the request header has ' +
+      'info attached', (done) => {
+      const request = {
+        headers: { },
+        session: { },
+      };
+      request.headers.authorization = collectorToken;
+      jwtUtil.verifyToken(request, dummyCallback)
+      .then(() => {
+        expect(request.headers.userName).to.equal(collectorInst.name);
+        expect(request.headers.profileName).to.equal(undefined);
+        expect(request.headers.tokenName).to.equal(collectorInst.name);
+        expect(request.headers.isAdmin).to.equal(false);
+        expect(request.headers.isBot).to.equal(false);
+        expect(request.headers.isCollector).to.equal(true);
+        return done();
+      }).catch(done);
+    });
+
+    it('verifyUserToken with admin user and make sure the request header has ' +
+      'info attached', (done) => {
+      const request = {
+        headers: { },
+        session: { },
+      };
+      request.headers.authorization = predefinedAdminUserToken;
+      jwtUtil.verifyToken(request, dummyCallback)
+      .then(() => {
+        expect(request.headers.userName).to.equal(adminUser.name);
+        expect(request.headers.profileName).to.equal('Admin');
+        expect(request.headers.tokenName).to.equal(adminUser.name);
+        expect(request.headers.isAdmin).to.equal(true);
+        expect(request.headers.isBot).to.equal(false);
+        expect(request.headers.isCollector).to.equal(false);
+        return done();
+      }).catch(done);
+    });
+
+    it('verifyToken with token added to session object', (done) => {
+      const request = {
+        headers: { },
+        session: { },
+      };
+      request.session.token = userToken;
+      jwtUtil.verifyToken(request, dummyCallback)
+      .then(() => {
+        expect(request.headers.userName).to.equal(userInst.name);
+        expect(request.headers.profileName).to.equal('___myProfile');
+        expect(request.headers.tokenName).to.equal('___myRefocusUser');
+        expect(request.headers.isAdmin).to.equal(false);
+        expect(request.headers.isBot).to.equal(false);
+        expect(request.headers.isCollector).to.equal(false);
+        return done();
+      }).catch(done);
+    });
+
+    it('verifyToken with invalid token', (done) => {
```

Comment:
I am a bit confused with this test - shouldn't this throw some kind of forbidden error because token is invalid?

Reply:
Yes, I expected it to throw an error too but it did not. So I convinced myself with the following argument. 
"When an error is thrown by verifyToken function, all we are doing it calling the callback with the error as an argument. The callback that I am using just resolves to true no matter what. The callback used by the "swaggerSecurity" middleware is probably sophisticated enough to handle all the cases" 
Does this make sense ?

Reply:
Yes, I think that makes sense. And we have corresponding test in tests/enforceToken/verbs.js. Thanks for explaining!